js/login.js

```diff
@@ -0,0 +1,107 @@
+const CLIENT_ID =
+  "822723435779-entgq7gcoh7lsobv2pjlg1ihbevfsio6.apps.googleusercontent.com";
+const API_KEY = "AIzaSyBGqMCLntJ40IGU9HU1objWRgDqQw1GXdw";
+
+const DISCOVERY_DOC =
+  "https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest";
+
+const SCOPES = "https://www.googleapis.com/auth/calendar";
+
+let tokenClient;
+let gapiInited = false;
+let gisInited = false;
+
+document.getElementById("authorize_button").style.display = "none";
+document.getElementById("signout_button").style.display = "none";
+
+function gapiLoaded() {
+  gapi.load("client", intializeGapiClient);
+}
+
+async function intializeGapiClient() {
+  await gapi.client.init({
+    apiKey: API_KEY,
+    discoveryDocs: [DISCOVERY_DOC],
+  });
+  gapiInited = true;
+  maybeEnableButtons();
+}
+
+function gisLoaded() {
+  tokenClient = google.accounts.oauth2.initTokenClient({
+    client_id: CLIENT_ID,
+    scope: SCOPES,
+    callback: "",
+  });
+  gisInited = true;
+  maybeEnableButtons();
+}
+
+function maybeEnableButtons() {
+  if (gapiInited && gisInited) {
+    document.getElementById("authorize_button").style.display = "unset";
+  }
+}
+
+function handleAuthClick() {
+  tokenClient.callback = async (resp) => {
+    if (resp.error !== undefined) {
+      throw resp;
+    }
+    document.getElementById("signout_button").style.display = "unset";
+    document.getElementsByClassName("main")[0].style.display = "grid";
+    document.getElementsByClassName("wrapper-logout")[0].style =
+      "display: none;";
+
+    document.getElementById("authorize_button").style.display = "none";
+  };
+
+  if (gapi.client.getToken() === null) {
+    tokenClient.requestAccessToken({ prompt: "consent" });
+  } else {
+    tokenClient.requestAccessToken({ prompt: "" });
+  }
+}
+
+function handleSignoutClick() {
+  const token = gapi.client.getToken();
+  if (token !== null) {
+    google.accounts.oauth2.revoke(token.access_token);
+    gapi.client.setToken("");
+    document.getElementById("authorize_button").style.display = "unset";
+    document.getElementById("signout_button").style.display = "none";
+    document.getElementsByClassName("main")[0].style.display = "none";
```

Comment:
this functions should be separated from ur api logic. The sense of modules is that u can import them wherever u want and use them, without any decencies

Reply:
Yes, 9 for this one.